src/sentry/tsdb/snuba.py

```diff
@@ -0,0 +1,167 @@
+
+        model_group, model_aggregate = model_columns
+
+        groupby = []
+        if group_on_model and model_group is not None:
+            groupby.append(model_group)
+        if group_on_time:
+            groupby.append('time')
+        if aggregation == 'count' and model_aggregate is not None:
+            # Special case, because count has different semantics, we change:
+            # `COUNT(model_aggregate)` to `COUNT() GROUP BY model_aggregate`
+            groupby.append(model_aggregate)
+            model_aggregate = None
+
+        keys_map = dict(zip(model_columns, self.flatten_keys(keys)))
+        keys_map = {k: v for k, v in six.iteritems(keys_map) if k is not None and v is not None}
+        if environment_id is not None:
+            keys_map['environment'] = [environment_id]
+
+        return snuba.query(start, end, groupby, None, keys_map,
+                           aggregation, model_aggregate, rollup)
+
+    def get_range(self, model, keys, start, end, rollup=None, environment_id=None):
+        result = self.get_data(model, keys, start, end, rollup, environment_id,
+                               aggregation='count', group_on_time=True)
+        # convert
+        #    {group:{timestamp:count, ...}}
+        # into
+        #    {group: [(timestamp, count), ...]}
+        return {k: sorted(result[k].items()) for k in result}
+
+    def get_distinct_counts_series(self, model, keys, start, end=None,
+                                   rollup=None, environment_id=None):
+        result = self.get_data(model, keys, start, end, rollup, environment_id,
+                               aggregation='uniq', group_on_time=True)
+        # convert
+        #    {group:{timestamp:count, ...}}
+        # into
+        #    {group: [(timestamp, count), ...]}
+        return {k: sorted(result[k].items()) for k in result}
+
+    def get_distinct_counts_totals(self, model, keys, start, end=None,
+                                   rollup=None, environment_id=None):
+        return self.get_data(model, keys, start, end, rollup, environment_id,
+                             aggregation='uniq')
+
+    def get_distinct_counts_union(self, model, keys, start, end=None,
+                                  rollup=None, environment_id=None):
+        return self.get_data(model, keys, start, end, rollup, environment_id,
+                             aggregation='uniq', group_on_model=False)
+
+    def get_most_frequent(self, model, keys, start, end=None,
+                          rollup=None, limit=10, environment_id=None):
+        aggregation = 'topK({})'.format(limit)
+        result = self.get_data(model, keys, start, end, rollup, environment_id,
+                               aggregation=aggregation)
+        # convert
+        #    {group:[top1, ...]}
+        # into
+        #    {group: [(top1, score), ...]}
+        for k in result:
+            item_scores = [(v, float(i + 1)) for i, v in enumerate(reversed(result[k]))]
```

Comment:
Is the enumeration ordinal just a placeholder since due to `topK` not returning the counts? I think that'd at least be worth a comment here clarifying this isn't a direct translation.

Reply:
Correct. I'll leave a comment.